api/config/observability.go

```diff
@@ -4,16 +4,16 @@ import "time"
 
 // ObservabilityPublisher
 type ObservabilityPublisher struct {
-	ArizeSink          ArizeSink
-	BigQuerySink       BigQuerySink
-	KafkaConsumer      KafkaConsumer
-	ImageName          string
-	DefaultResources   ResourceRequestsLimits
-	EnvironmentName    string
-	Replicas           int32
-	TargetNamespace    string
-	ServiceAccountName string
-	DeploymentTimeout  time.Duration `default:"30m"`
+	ArizeSink                ArizeSink
+	BigQuerySink             BigQuerySink
+	KafkaConsumer            KafkaConsumer
+	ImageName                string
+	DefaultResources         ResourceRequestsLimits
+	EnvironmentName          string
+	Replicas                 int32
+	TargetNamespace          string
+	DeploymentTimeout        time.Duration `default:"30m"`
+	ServiceAccountSecretName string
```

Comment:
I guess we're completely removing the ability for the platform maintainers of the Merlin API server to have the observability worker containers use workload identity? Just wanted to confirm if we've decided to completely replace credential access via workload identity with mounted GCP secrets (as opposed to giving the platform maintainers an option between the two).

Reply:
yes for now, we can add this again into shared library that can enrichPodIdentity based on some config e.g cloud provider, workload identity flag